tests/Dfe.PlanTech.Web.E2ETests/cypress/e2e/dynamic-page-validator/dpv2-recs-maturity-paths.cy.js

```diff
@@ -0,0 +1,49 @@
+import DataMapper from "export-processor/data-mapper.js";
+import { selfAssessmentSlug } from "./helpers/index.js";
+import { quickNavigateToRecommendations } from "./validators/index.js";
+
+
+const dataMapper = new DataMapper(require('../../fixtures/contentful-data'));
+
+describe("Recommendations", { testIsolation: false }, () => {
+
+    before(() => {
+        cy.loginWithEnv(`${selfAssessmentSlug}`);
+    });
+
+    (dataMapper?.mappedSections ?? []).forEach((section) => {
+        describe(`${section.name} recommendations`, { testIsolation: false }, () => {
+
+            // Establish section status using self-assessment page tag
+            let inProgress = false;
+            before(() => {
```

Comment:
I also think that if you moved this up a layer, into the 
```js
describe("Recommendations", { testIsolation: false }, () => {

    before(() => {
        cy.loginWithEnv(`${selfAssessmentSlug}`);
    });
```

section, you could modify it so that it completely skips creating the tests for that section anyway.

_However_, can we not just... reset the section? I.e. if it's in progress, just click through it, submit it, then run the tests? Save skipping them?

Reply:
Re beforeEach - it's complicated because a) Cypress and b) we're generating section tests in a forEach.
Replacing the before with beforeEach in situ applies it to the individual tests within that section, effectively re-isolating them. If we lift the hook out of the forEach loop and replace with a beforeEach we don't have a section to apply the check to. So I think before at section level is appropriate, and the higher before is so we're not trying to log in before each section (and failing).

Re the section reset - yes that's the goal and there's a ticket to do that (there's a link in the PR description somewhere I think), the skip is just a holdover until that's done but the hook gives us the scaffolding and will make that one easier when we get to it.

Reply:
You've raised some good points that honestly are going over my head.

However, how about this instead then

```js
          before(function () {
              cy.visit(`${selfAssessmentSlug}`);

              cy.get("a.govuk-link")
                  .contains(section.name.trim())
                  .parent()
                  .next()
                  .within(() => {
                      cy.get("strong.app-task-list__tag")
                          .invoke("text")
                          .then((text) => {
                              const inProgress = text.includes("in progress");

                              if (inProgress) {
                                  console.log(`Skipping all tests for section: ${section.name} (status is 'in progress')`);
                                  this.skip();
                              }
                          });
                  });
          });
```

Tested this locally and seems to work, and just combines the two befores basically. Looks cleaner to me, but YMMV. Please feel free to tell me you prefer the current implementation and that I can go away etc.

Reply:
I kind of did this already, but with a command, see latest :)

Reply:
better!